### ISSUE TITLE
Sporadic error message on Blinker 1.8+

### ISSUE THREAD

**Issue by mpaglia0** [pelican-plugins/sitemap#35]
Compiling Pelican 4.10 (Python 3.11) I get this error message:

```python
Exception ignored in: <function WeakMethod.__new__.<locals>._cb at 0x7f096ba1e3e0>
Traceback (most recent call last):
  File "/usr/lib64/python3.11/weakref.py", line 57, in _cb
AttributeError: 'NoneType' object has no attribute '_alive'
Exception ignored in: <function WeakMethod.__new__.<locals>._cb at 0x7f096ba1e840>
Traceback (most recent call last):
  File "/usr/lib64/python3.11/weakref.py", line 57, in _cb
AttributeError: 'NoneType' object has no attribute '_alive'
Exception ignored in: <function WeakMethod.__new__.<locals>._cb at 0x7f096ba1eca0>
Traceback (most recent call last):
  File "/usr/lib64/python3.11/weakref.py", line 57, in _cb
AttributeError: 'NoneType' object has no attribute '_alive'
```

**Comment by justinmayer** [comment]
Ciao Maurizio. I can't reproduce this error. In order to provide assistance, it is important to adhere to the steps in Pelican's [How to Get Help](https://docs.getpelican.com/en/latest/contribute.html#how-to-get-help) documentation. Otherwise there is really no way to offer any help or guidance. Would you please follow the steps in those instructions and then post a detailed update here?

**Comment by projectgus** [comment]
I haven't managed to come up with a simple reproducer, but I'm pretty sure this issue is coming from version 1.8.2 of the `blinker` dependency of the sitemap plugin. More details at the issue linked above.

I found a quick workaround was to roll back to blinker 1.7.0.

**Comment by justinmayer** [comment]
@projectgus: Many thanks for the detailed report of your investigation into this issue. Hopefully that problem can be addressed upstream, and in the interim you have provided a helpful workaround. Much appreciated! ❤️

**Comment by projectgus** [comment]
> Side note… As far as I recall, Blinker is a _Pelican_ dependency 

Oh, you're of course right! It just happened that in my setup the only weakrefs I saw in pdb were being created by this plugin. Interesting.

>  If the problem is not addressed upstream in a patched Blinker release sometime soon, we could consider pinning Blinker to 1.7 and releasing Pelican 4.10.2 as an interim workaround.

Good to know, thanks. FWIW, at least for me I don't see any impact on functionality: Pelican finishes generating the site fine, and the "Exception ignored" messages are only printed as the Python interpreter is doing its final shutdown before exiting. The process exit code is still 0, etc.

**Comment by justinmayer** [comment]
Because the issue only seems (for now, anyway) to be affecting this plugin, for the moment I think I will take the liberty of pinning Blinker in the plugin's dependency list: https://github.com/pelican-plugins/sitemap/commit/70ae00896412de1c3dd1f842c19725f998347f13

@projectgus: Since you are a Sitemap plugin user — and also a seemingly very competent sleuth! 😉 — if I might enlist your assistance, there have been some [errors when running Sitemap plugin tests](https://github.com/pelican-plugins/sitemap/issues/36), but only in the GitHub CI environment — I cannot replicate these errors locally. Any chance you could take a look and post any insights into the issue I just created to track this? #36

**Comment by projectgus** [comment]
> Any chance you could take a look and post any insights into the issue I just created to track this? #36

Happy to take a look, turns out the same failure shows up on my Linux system. Have opened #37 with a possible fix.

Thanks for all your work on Pelican!

**Comment by justinmayer** [comment]
Thank you for the kind words, Angus. Much appreciated ❤️ 

I released your fix, among other changes, in [Sitemap 1.2.0](https://github.com/pelican-plugins/sitemap/releases/tag/1.2.0) ✨
